zebra/zebra_vxlan.c

```diff
@@ -3427,6 +3427,19 @@ static void zvni_mac_del_hash_entry(struct hash_bucket *bucket, void *arg)
 			zvni_mac_uninstall(wctx->zvni, mac);
 
 		zvni_mac_del(wctx->zvni, mac);
+	} else if ((wctx->flags & DEL_LOCAL_MAC)
```

Comment:
Two questions.
1.  why the existing if check will be true for this case ? I specifically mean "(wctx->flags & DEL_LOCAL_MAC) && (mac->flags & ZEBRA_MAC_LOCAL)". Is this because mac->flags is not set with "ZEBRA_MAC_LOCAL" for AUTO case ?
2. Do you think it makes sense to add this check to existing if check instead of another else-if ? The existing if check also handles deleting from client, mac_uninstall etc. Are these two not required for Auto case ?

Reply:
Ans. 1: Yes since the MAC is AUTO state which is a special case where it hangs around due to association with Neighbor entries. 

Ans. 2: The existing check has uninstall flag set and already has many conditions. The auto mac is a special case which just need to remove the auto mac from vni db.